core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/service/contractdefinition/ContractDefinitionQueryValidator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ *  Copyright (c) 2023 SAP Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       SAP Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.connector.service.contractdefinition;
+
+import org.eclipse.edc.connector.contract.spi.types.offer.ContractDefinition;
+import org.eclipse.edc.connector.service.query.QueryValidator;
+import org.eclipse.edc.spi.result.Result;
+
+import java.util.regex.Pattern;
+
+
+/**
+ * Validates that a particular ContractDefinition query is valid,
+ * i.e. when query left-hand operand does not contain string "privateProperties" then
+ * conforms is it in the canonical format,
+ * If it contains privateProperties then just validate whether pattern is correct.
+ */
+public class ContractDefinitionQueryValidator extends QueryValidator {
```

Comment:
What is this validator meant to? If the idea is to avoid particular characters then we should tackle it in a more holistic way for all the queries (eventually in a different PR).

Reply:
Changes are similar to Asset.
https://github.com/eclipse-edc/Connector/blob/main/core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/service/asset/AssetQueryValidator.java

EDC_NAMESPACE/privateProperties.'Key'  path is not valid from base class QueryValidator perspective.
Here key can be be anything and hence can't follow static path element.

Reply:
the Asset query validator works in a different way because on the asset there's the possibility to filter them by the "public" properties without specifying the `properties.` prefix.

If `QueryValidator` cannot handle `EDC_NAMESPACE + "privateProperties.'Key'` I think that needs to be fixed there and not with an extension (that will be the same for pretty much all the queryable entities).

I mean, I guess that map query were disabled because of the possibility to inject malicious code as the structure of the map is not defined at compile time, but I think it could be handled in a different way (like validating the json path passed)
what's the point to avoid query on map fields if then we permit it through extensions?

Reply:
Done with changes. Now jsonpath validated in QueryValidator